go/coredns/plugin/pfdns/pfdns.go

```diff
@@ -263,7 +270,7 @@ func (pf *pfdns) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg)
 					}
 				}
 				log.LoggerWContext(ctx).Debug("Get answer from pffilter for " + state.QName())
-				pf.DNSFilter.Set(mac+state.QName(), answer, cache.DefaultExpiration)
+				pf.DNSFilter.Set(mac+category+strconv.FormatBool(violation)+state.QName(), answer, cache.DefaultExpiration)
```

Comment:
Just need a helper function to create the cache key so that if we ever change the creation, we don't need to remember all the places to change